tardis/energy_input/gamma_ray_packet_source.py

```diff
@@ -737,28 +777,23 @@ def create_packets(
 
         # sample radii at time = 0
         initial_radii = self.create_packet_radii(sampled_packets_df)
+        # sample decay times
+        sampled_times = sampled_packets_df.index.get_level_values("time") * (
+            u.d
+        ).to(u.s)
 
+        # TODO: Rewrite this without for loop. This is expensive
         # get the time step index of the packets
-        initial_time_indexes = sampled_packets_df.index.get_level_values(0)
-
-        # get the time of the middle of the step for each packet
-        packet_effective_times = np.array(
-            [self.effective_times[i] for i in initial_time_indexes]
-        )
-
-        # packet decay time
-        times = self.create_packet_times_uniform_energy(
-            number_of_packets,
-            sampled_packets_df.index.get_level_values(2),
-            packet_effective_times,
-        )
+        decay_time_indices = []
+        for i in range(number_of_packets):
```

Comment:
If the decay time indices is not needed, perhaps just remove this part of the code?

Reply:
Yes. True. Added `time_index` as part of the G_packet. This eliminates the use of the for loop also.